client/src/main.rs

```diff
@@ -199,7 +204,8 @@ pub async fn main() -> Result<(), Box<dyn Error>> {
         } else if let Some(_matches) = matches.subcommand_matches("edge-property") {
             unimplemented!();
         }
+    } else {
+        panic!("unknown command");
     }
-
-    panic!("unknown command");
```

Comment:
This old panic should still be here, since at the moment a failed match on any of the nested ifs won't panic.

Reply:
Got it. 
I also found a [setting](https://docs.rs/clap/2.32.0/clap/enum.AppSettings.html#variant.SubcommandRequiredElseHelp) for the App/Subcommand which enforces that at least one of the subcommands *must* be used.

I think this will error out more gracefully than panicking manually.

Reply:
Have pushed a few more changes. 
> If match fails on any of the nested ifs, a panic won't occur with this change.

By enforcing that "at least one" of the nested subcommands is used at runtime (using `AppSettings::SubcommandRequiredElseHelp`), we should not run into this condition

Reply:
Oh cool, nice find!